modular_zubbers/code/modules/wiremod/components/sensors/dronecam.dm

```diff
@@ -0,0 +1,129 @@
+	var/datum/port/input/on
+	/// Camera range flag (short/full)
+	var/datum/port/input/full_range
+	/// The network to use
+	var/datum/port/input/network
+
+	/// Camera range internal flag
+	var/full_range_current
+	/// Camera random ID
+	var/c_tag_random
+
+	var/mob/living/circuit_drone/drone
+
+/obj/item/circuit_component/drone_camera/get_ui_notices()
+	. = ..()
+	. += create_ui_notice("Power Usage For Short Range: [power_usage_per_input] Per [DisplayTimeText(COMP_CLOCK_DELAY)]", "orange", "clock")
+	. += create_ui_notice("Power Usage For Full Range: [power_usage_per_input_full_range] Per [DisplayTimeText(COMP_CLOCK_DELAY)]", "orange", "clock")
+
+/obj/item/circuit_component/drone_camera/populate_ports()
+	on = add_input_port("On", PORT_TYPE_NUMBER, default = 0)
+	full_range = add_input_port("Full Range", PORT_TYPE_NUMBER, default = 0)
+	network = add_input_port("Network", PORT_TYPE_STRING, default = "ss13")
+
+	full_range_current = full_range.value
+	c_tag_random = rand(1, 999)
+
+/obj/item/circuit_component/drone_camera/pre_input_received(datum/port/input/port)
+	if(drone?.shell_camera)
+		if(!on.value)
+			stop_process()
+			if(drone.shell_camera.status) //Instantly turn off the camera
+				drone.shell_camera.toggle_cam(null, 0)
+		else if(on.value)
+			update_name_network(drone)
+			start_process()
+
+/obj/item/circuit_component/drone_camera/Destroy()
+	stop_process()
+	if(drone?.shell_camera)
+		if(drone.shell_camera.status)
+			drone.shell_camera.toggle_cam(null, 0)
+	return ..()
+
+/obj/item/circuit_component/drone_camera/register_shell(atom/movable/shell)
+	stop_process()
+	. = ..()
+	if(istype(shell, /mob/living/circuit_drone))
+		drone = shell
+		drone.shell_camera = new /obj/machinery/camera/drone (drone)
+		drone.shell_camera.toggle_cam(null, 0)
+		drone.shell_camera.setViewRange(full_range.value > 0 ? 7 : 2)
+		full_range_current = full_range.value
+		update_name_network(drone)
+
+/obj/item/circuit_component/drone_camera/unregister_shell(atom/movable/shell)
+	stop_process()
+	if(drone?.shell_camera)
+		if(drone.shell_camera.status)
+			drone.shell_camera.toggle_cam(null, 0)
+		QDEL_NULL(drone.shell_camera)
+	drone = null
+	. = ..()
+
+/obj/item/circuit_component/drone_camera/process(seconds_per_tick)
+	if(drone?.shell_camera)
+		if(drone.health < 0) //If shell is destroyed
+			if(drone.shell_camera.status) //Turn off camera
+				drone.shell_camera.toggle_cam(null, 0)
+			return
+		var/obj/item/stock_parts/cell/cell = parent.get_cell()
+		if(!cell || !cell?.use(full_range.value > 0 ? power_usage_per_input_full_range : power_usage_per_input)) //If cell doesn't exist, or we ran out of power
+			if(drone.shell_camera.status) //Turn off camera
+				drone.shell_camera.toggle_cam(null, 0)
+			return
+		if(!full_range.value != !full_range_current) //If the camera range has changed, update camera range
+			drone.shell_camera.setViewRange(full_range.value > 0 ? 7 : 2)
+			full_range_current = full_range.value
```

Comment:
Defines needed again. You might be better off putting both camera components into one file so you can `#undef` in the same file